atom/browser/api/atom_api_app.cc

```diff
@@ -268,6 +269,30 @@ v8::Local<v8::Value> App::DefaultSession(v8::Isolate* isolate) {
     return v8::Local<v8::Value>::New(isolate, default_session_);
 }
 
+bool App::MakeSingleInstance(v8::Local<v8::Function> callback) {
```

Comment:
When I did that, I had V8 reference issues (i.e. the callback would be GCd and we'd end up jumping off into nowhere)